### PR TITLE
Added new ui tests

### PR DESCRIPTION
**Pull request by jrbenny35**
Adds 4 new tests.
```test_devhub_logout```, Tests logging out from the developer hub.
```test_devhub_register```, Tests the register links navigates to the fxa signup page.
```test_title_routes_to_home```, tests Firefox Addons title routes to home page.
```test_more_dropdown_navigates_correctly```, tests the more drop down navigates to the correct links.

### CODE REVIEW

File: tests/ui/pages/desktop/base.py [hunk 1]
```diff
@@ -90,6 +100,20 @@ def click_logout(self):
...
+        action = ActionChains(self.selenium)
+        action.move_to_element(menu)
+        action.pause(2)
+        action.click()
+        action.pause(2)
+        action.move_to_element(links[item])
+        action.pause(2)
+        action.click()
+        action.pause(2)
+        action.perform()
```
Comment (EnTeQuAk) [inline tests/ui/pages/desktop/base.py]: Can we add a few comments for the separate steps what which step does? Simply reading this method makes it really hard to follow along.

File: tests/ui/pages/desktop/devhub.py [hunk 2]
```diff
@@ -53,10 +54,13 @@ def login(self, email, password):
...
             lambda _: self.is_element_displayed(*self._avatar_locator)
         )
+        return self
 
     @property
     def logged_in(self):
-        return self.is_element_displayed(*self.header._sign_in_locator)
+        if self.is_element_displayed(*self.header._sign_in_locator):
+            return False
+        return True
```
Comment (EnTeQuAk) [inline tests/ui/pages/desktop/devhub.py]: What changed? Doesn't the previous `return` do exactly the same?

File: tests/ui/test_devhub.py [hunk 3]
```diff
@@ -33,3 +35,26 @@ def test_devhub_addon_edit_link_works(base_url, selenium, devhub_login):
...
+    devhub = devhub_login.header.click_sign_out()
+    assert not devhub.logged_in
+
+
+@pytest.mark.desktop_only
+@pytest.mark.nondestructive
+@pytest.mark.withoutresponses
+def test_devhub_register(base_url, selenium):
+    """Test register link loads register page."""
+    selenium.get('http://olympia.test/en-US/developers/')
```
Comment (EnTeQuAk) [inline tests/ui/test_devhub.py]: Why are you requesting this resource? This doesn't return anything or doesn't have any side effects.

File: tests/ui/pages/desktop/devhub.py (hunk 2 above)
Comment (jrbenny35) [reply tests/ui/pages/desktop/devhub.py]: The logic goes: If the sign in locator is displayed, you are not ```logged_in```. If it is not displayed then you are. Just using ```is_displayed```  didn't correlate logically to how the property would be used.

File: tests/ui/test_devhub.py (hunk 3 above)
Comment (jrbenny35) [reply tests/ui/test_devhub.py]: Probably a copy/paste error.

File: tests/ui/pages/desktop/devhub.py (hunk 2 above)
Comment (EnTeQuAk) [reply tests/ui/pages/desktop/devhub.py]: Ah, you could du `return not self.is_element_displayed(*self.header._sign_in_locator)` then I'd guess?

Comment (jrbenny35) [reply tests/ui/pages/desktop/devhub.py]: Ah, yeah, being all pythonic and stuff.
